chall04/gbudau.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+
+import sys
+
+STDIN = 0
+FILE = 1
+
+def read_error(filename):
+    print("{}: {}: Can't read file".format(sys.argv[0], filename))
+
+def space_error(filename):
+    print("{}: {}: Not enough space in the given shelves".format(sys.argv[0], filename))
+
+def fill_bookshelves(filename, bookshelves, books):
+    # Convert books and bookshelves to integers
+    # Sort from max to min
+    books = [int(i) for i in books]
+    bookshelves = [int(i) for i in bookshelves]
+    books.sort(reverse = True)
+    bookshelves.sort(reverse = True)
+
+    used_bookshelves = []
+    while books and bookshelves and ((bookshelves[0] >= books[0]) or (used_bookshelves and used_bookshelves[0] >= books[0])):
+        if  used_bookshelves and used_bookshelves[0] >= books[0]:
+            used_bookshelves[0] -= books[0]
+            books.pop(0)
+            used_bookshelves.sort(reverse = True)
+        else:
+            used_bookshelves.append(bookshelves[0] - books[0])
+            used_bookshelves.sort(reverse = True)
+            books.pop(0)
+            bookshelves.pop(0)
+
+    space_error(filename) if books else print(len(used_bookshelves))
+
+def verify_and_save_file(arg, source):
+    bookshelves = []
+    books = []
+
+    # Save first line to bookshelves list from the file generator, striping spaces and newline
+    # Splitting the line into a list of numeric strings
+    # next() can return an error when the file is empty, without any newline so im using try/except
+    # Also checking the shelves to have non-zero values
+    try:
+        bookshelves = next(arg).rstrip().split()
```

Comment:
You do not need to use rstrip if you use split without parameters:
```suggestion
    bookshelves = next(arg).split()
```
As stated in the [method's documentation](https://docs.python.org/3/library/stdtypes.html#str.split):
> If sep is not specified or is None, a different splitting algorithm is applied: runs of consecutive whitespace are regarded as a single separator, and the result will contain no empty strings at the start or end if the string has leading or trailing whitespace. Consequently, splitting an empty string or a string consisting of just whitespace with a None separator returns [].

Reply:
Thank you, i didn't knew about this.